modules/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/tree/BLangImportPackage.java

```diff
@@ -93,7 +93,11 @@ public boolean equals(Object o) {
             return false;
         }
 
-        return version != null ? version.equals(that.version) : that.version == null;
+        if (version != null ? !version.equals(that.version) : that.version != null) {
+            return false;
+        }
+
+        return alias.equals(that.alias);
```

Comment:
we shouldn't match alias right?

Reply:
Otherwise, if there are two or more imports (same package) with different aliases (see below), only one of them will get included in the tree.
```
import ballerina.lang.system as x;
import ballerina.lang.system as y;
```

But we need to include them in the tree, but validate later during the semantic analyzer phase.

Reply:
ok got it.